### ISSUE TITLE
SIGABRT errors with complex structures

### ISSUE THREAD

**Issue by Fuyukai** [ccubed/Earl#7]
```
In [1]: import earl

In [2]: obb = {"d": {"v": 6, "shard": [0, 1], "compress": False, "token": "", "large_threshold": 250, "properties": {"$os": "linux", "$device": "curious", "$browser": "curious", "$referrer": "", "$referring_domain
   ...: ": ""}}, "op": 2}

In [3]: earl.pack(obb)
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
fish: “ipython” terminated by signal SIGABRT (Abort)```

**Comment by ccubed** [comment]
Initial thoughts: I'm returning bad data.
Test: 
```python
In [1]: import earl
In [2]: earl.pack({"d": {"v": 6, "shard": [0, 1], "compress": False, "token": ""}})
```
Resulted in abort.

So this fails at packing.

Second thought: "" is interpreted as null.
Test:
```python
In [1]: import earl
In [2]: earl.pack({"d": ""})
```
Resulted in abort.

So yes, that's our issue. Reading "" results in only the Null Terminator being read. The issue now is how to return a blank string in python's C API.

**Comment by ccubed** [comment]
I know why this is broken now. Python wasn't stopping at the error. Thus it was attempting to push a NULL to the buffer. Though what to do to fix that is another issue.

**Comment by ccubed** [comment]
Fixed as of f90b90d347b65c30c76079c162c9059c6a530209. Unpacking now broken.
